logpy/variable.py

```diff
@@ -1,78 +0,0 @@
-from contextlib import contextmanager
-from .util import hashable
-from .dispatch import dispatch
-
-_global_logic_variables = set()
-_glv = _global_logic_variables
-
-
-class Var(object):
-    """ Logic Variable """
-    __slots__ = ('token', )
-
-    _id = 1
-
-    def __init__(self, token=None):
-        if token is None:
-            token = "_%s" % Var._id
-            Var._id += 1
-        self.token = token
-
-    def __str__(self):
-        return "~" + str(self.token)
-
-    __repr__ = __str__
-
-    def __eq__(self, other):
-        return type(self) == type(other) and self.token == other.token
-
-    def __hash__(self):
-        return hash((type(self), self.token))
-
-
-var = lambda *args: Var(*args)
-vars = lambda n: [var() for i in range(n)]
-
-
-@dispatch(Var)
-def isvar(v):
-    return True
-
-
-@dispatch(object)
-def isvar(o):
-    return not not _glv and hashable(o) and o in _glv
-
-
-@contextmanager
-def variables(*variables):
```

Comment:
It seems this context manager is designed to replace minikanren's `fresh`? 

Since python has no symbol type, it seems like it muddles the notions of equality. Unifying against a string in the presence of a global logic variable doesn't seem like it has clean semantics. This context manager also does not appear to be threadsafe, so it might make sense to eliminate it or de-emphasize it in the `unification` library.